In [1]:
import pandas as pd

In [18]:
df = pd.read_csv('/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/raceInProgress/raceV2.csv')

In [19]:
df.head()

PcrKey  ...  nativeHawaiian/otherPacificIslander
0  13273412  ...                                    0
1  17184298  ...                                    0
2  21699657  ...                                    0
3  72057010  ...                                    0
4  83733589  ...                                    0

[5 rows x 7 columns]

In [20]:
len(df.index)

33438658

In [25]:
print(df.PcrKey.nunique())

32866360


- This was the exact count we had after throwing out the nans so we're good here
- We still have some duplicate PcrKeys in here that we need to drop

In [21]:
dup_first = df[df.duplicated('PcrKey')]
dup_first.head()

PcrKey  ...  nativeHawaiian/otherPacificIslander
55   118170967  ...                                    0
212  118171544  ...                                    0
493  118172608  ...                                    0
668  118173589  ...                                    0
743  118200233  ...                                    0

[5 rows x 7 columns]

In [22]:
len(dup_first.index)

572298

In [23]:
dup = df[df.duplicated('PcrKey',False)]

In [24]:
print(len(dup))
print(dup.PcrKey.nunique())

1064494
492196


In [6]:
dup.head()

PcrKey  ...  nativeHawaiian/otherPacificIslander
54   118170967  ...                                    0
55   118170967  ...                                    0
211  118171544  ...                                    1
212  118171544  ...                                    0
492  118172608  ...                                    0

[5 rows x 7 columns]

- Now we're gonna have to merge these
    - Group by the pcr key and then just sum all of the other values

In [12]:
df = df.groupby(['PcrKey'],as_index=False).agg('sum')

In [13]:
df.head()

PcrKey  ...  nativeHawaiian/otherPacificIslander
0  13273412  ...                                    0
1  17184298  ...                                    0
2  21699657  ...                                    0
3  72057010  ...                                    0
4  83733589  ...                                    0

[5 rows x 7 columns]

In [14]:
len(df.index)

32866360

- Started with 49,555,288 rows
- Threw out 16,116,630 rows with missing data
    - Leaving me with 33,438,658 rows
- Then we merged the non-unique PCR rows and left with a CORRECT total number of unique PCRs = 32,866,360

In [16]:
# pretty sure this is correct but we need to reload the original df to check so lets just exportt
df.to_csv('/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/raceInProgress/raceV3.csv',index=False)